app/views/users/_main_feed.html.erb

```diff
@@ -14,10 +14,18 @@
 <% if @stories.present? %>
   <%= render "articles/single_story", story: @stories.first, featured: false %>
   <% if @comments.present? %>
-    <%= render "users/comments_section" %>
+    <% if user_signed_in? %>
+      <%= render "users/comments_section" %>
+    <% else %>
+      <%= render "users/comments_locked_cta" %>
+    <% end %>
   <% end %>
   <%= render partial: "articles/single_story", collection: @stories[1, @stories.size], as: :story, locals: { featured: false } %>
   <div class="placeholder-div"></div>
 <% elsif @comments.present? %>
-  <%= render "users/comments_section" %>
```

Comment:
This `comments_section` is getting used at [here](https://github.com/forem/forem/blob/c60b41f025f16c13f532bcbe81ed0ef8b1c67cd3/app/views/users/show.html.erb#L132) too but I am unable to understand under what situations this will be shown.

This information is needed so that I can decide whether I should display `comments_locked_cta` or not?

Reply:
I'm just taking a guess here but this file dir would indicate it's on its feed on a user's profile page. I think it's for how the rendering would look if a user doesn't a single article at all.

Reply:
Okay, in that case it means that user is signed-in, which in turn mean that we should keep it as it is.

This PR shows the changed UI only for logged-out users.

Thanks @maestromac

Reply:
this is just anyone's profile page, ie dev.to/maestromac. whether or not the user is signed in doesn't affect the rendering choice here.

Reply:
Thanks. I have applied the required change for [here](https://github.com/forem/forem/blob/c60b41f025f16c13f532bcbe81ed0ef8b1c67cd3/app/views/users/show.html.erb#L132) screen too.

Doubt has been resolved.